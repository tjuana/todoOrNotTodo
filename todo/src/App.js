import React from 'react';
import logo from './logo.svg';
import "./styles.css";
import Task from "./components/Task";
import TaskInput from "./components/TaskInput";


class App extends React.Component {
  constructor() {
	super();

	///real part
	

	console.log(localStorage.getItem('todo'));
    this.state = {
      tasks: [
	  ]
	};
	this.state.tasks = JSON.parse(localStorage.getItem('todo'));
  }
 
  toLocal = (tasks) => {
	localStorage.setItem('todo', JSON.stringify(this.state.tasks));
	};

  editTaskView = (id) => {
	const index = this.state.tasks.map((task) => task.id).indexOf(id);
	const { tasks } = this.state;
	const newTasks = tasks;
	if (newTasks[index].editing === false) {
		newTasks[index].editing = true;
		
		} else {
			newTasks[index].editing = false;
		}
	
	
    this.setState(tasks => newTasks);
  };

  addTask = (task) => {
    const { tasks } = this.state;
    const newTasks = tasks;
    newTasks.push({
      id: newTasks.length !== 0 ? newTasks.length : 0,
      title: task,
      done: false,
      editing: false
    });
	this.setState((tasks) => newTasks);
	this.toLocal(newTasks);
  };

  doneTask = (id) => {
    const index = this.state.tasks.map((task) => task.id).indexOf(id);
    this.setState((state) => {
      let { tasks } = this.state;
      tasks[index].done = true;
      return tasks;
    });
  };

  deleteTask = (id) => {
	const index = this.state.tasks.map((task) => task.id).indexOf(id);
    this.setState((state) => {
      let { tasks } = state;
      delete tasks[index];
      return tasks;
    });
  };

  editTaskInput = (id, input) => {
	const {tasks} = this.state;
	const newTasks = tasks;
	const index = this.state.tasks.map((task) => task.id).indexOf(id);

	newTasks[index].title = input;
	this.setState((tasks) => newTasks);
	};

  render() {
    const { tasks } = this.state;
    const activeTasks = tasks.filter((task) => !task.done);
    const doneTasks = tasks.filter((task) => task.done);
    return (
      <div className="App">
        <h1 className="top">{activeTasks.length} tasks ToDo {new Date().toLocaleDateString()}</h1>
        {[...activeTasks, ...doneTasks].map((task) => (
          <Task
            doneTask={() => this.doneTask(task.id)}
            deleteTask={() => this.deleteTask(task.id)}
			editTaskView={() => this.editTaskView(task.id)}
			editTaskInput={this.editTaskInput}
            task={task}
            key={task.id}
            edit={task.editing}
          ></Task>
        ))}
        <TaskInput addTask={this.addTask}></TaskInput>
		<img src={logo}></img>
      </div>

    );
  }
}

export default App;
