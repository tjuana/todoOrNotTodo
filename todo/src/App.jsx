import React from 'react';
import { Provider } from 'react-redux';
import { store } from './store/configureStore.jsx';
import Todo from './contaner/Todo.jsx';

const App = () => (
  <>
    <Provider store={store}>
      <Todo />
    </Provider>
  </>
);

export default App;

export { App as AppSnap };
