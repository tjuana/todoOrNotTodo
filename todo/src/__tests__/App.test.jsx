import React from 'react';
import renderer from 'react-test-renderer';
import { AppSnap } from '../App.jsx';

describe('Todo component test', () => {
  test('should snapshot', () => {
    const comp = renderer.create(<AppSnap />);
    let tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
