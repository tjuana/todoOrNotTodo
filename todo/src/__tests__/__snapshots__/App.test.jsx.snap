// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component test should  1`] = `
<div
  className="app"
>
  <h1
    className="top"
  >
    0
     
    TaSkS ToDo
     
    2020-9-7
  </h1>
  <div
    className="taskBlock"
  >
    <input
      onChange={[Function]}
      value=""
    />
    <button
      onClick={[Function]}
      type="button"
    >
      ADD...
    </button>
  </div>
</div>
`;
